client/extensions/woocommerce/woocommerce-services/components/token-field/index.js

```diff
@@ -0,0 +1,83 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import React from 'react';
+import PropTypes from 'prop-types';
+import { map, forEach } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import FormFieldset from 'components/forms/form-fieldset';
+import FormLabel from 'components/forms/form-label';
+import FormToggleInput from 'components/token-field';
+import FieldError from '../field-error';
+import FormSettingExplanation from 'components/forms/form-setting-explanation';
+
+export default class TokenField extends React.Component {
+	static propTypes = {
+		id: PropTypes.string.isRequired,
+		title: PropTypes.string,
+		description: PropTypes.string,
+		value: PropTypes.array.isRequired,
+		updateValue: PropTypes.func,
+		error: PropTypes.oneOfType( [ PropTypes.string, PropTypes.bool ] ),
+		className: PropTypes.string,
+		options: PropTypes.array,
+	};
+
+	render() {
+		const { id, title, description, value, placeholder, error, className, options } = this.props;
+
+		return (
+			<FormFieldset className={ className }>
+				<FormLabel htmlFor={ id }>{ title }</FormLabel>
+				<FormToggleInput
+					id={ id }
+					name={ id }
+					placeholder={ placeholder }
+					value={ value.map( selected => selected + '' ) }
+					suggestions={ map( options, option => option.id + '' ) }
```

Comment:
I believe if you provide the string names of the shipping classes here, partial matching while typing will occur. See "Default TokenField with Suggestions" here: https://wpcalypso.wordpress.com/devdocs/design/token-fields.

Reply:
I was not using class names here because they are not unique. Now however, I switched to slugs (which are normally very similar) and the partial matching is happening, even if it is not 100% accurate.

Reply:
I think it's safe to use the names even though they aren't unique. The core WooCommerce UI for managing shipping classes prevents duplicate names unless slugs are manually entered - so I think name collisions will be an edge case. Does the `TokenField` suggestion completion show the slug value or the name?

Reply:
<img width="449" alt="screen shot 2018-08-10 at 14 51 31" src="https://user-images.githubusercontent.com/5311119/43956439-e9653f5c-9cac-11e8-9eca-cbd6577809b8.png">

In the screenshot I am trying to autocomplete the "Book" shipping class, which has `book` as its slug. Still, this was how it was working before the change I just did. 

Now I am using actual names instead of the slugs and this allows autocompletion for non-latin strings, ex. "Книга" (Bulgarian for book, whose auto-generated slug is `
%d0%ba%d0%bd%d0%b8%d0%b3%d0%b0`).